components/webui/imports/ui/SearchView/SearchResults.jsx

```diff
@@ -49,10 +49,15 @@ const SearchResults = ({
     const hasMoreResults = visibleSearchResultsLimit < numResultsOnServer;
 
     const handleLoadMoreResults = useCallback(() => {
-        setVisibleSearchResultsLimit(
-            (v) => (v + VISIBLE_RESULTS_LIMIT_INCREMENT)
-        );
-    }, []);
+        if (hasMoreResults) {
+            setVisibleSearchResultsLimit(
+                visibleSearchResultsLimit + VISIBLE_RESULTS_LIMIT_INCREMENT
+            );
+        }
+    }, [
+        hasMoreResults,
+        visibleSearchResultsLimit,
+    ]);
```

Comment:
```suggestion
        if (hasMoreResults) {
            setVisibleSearchResultsLimit((v) => v + VISIBLE_RESULTS_LIMIT_INCREMENT);
        }
    }, [
        hasMoreResults,
        setVisibleSearchResultsLimit,
    ]);
```

Reply:
I feel this is slightly more appropriate since we should update the callback if `setVisibleSearchResultsLimit` changes.

Reply:
I agree. In fact it is not too meaningful to include both `hasMoreResults` and `visibleSearchResultsLimit` into the dependency array given any change in `visibleSearchResultsLimit` would also change `hasMoreResults`.